docs/src/tutorials/software-management.md

```diff
@@ -1,67 +1,38 @@
+# Manage from Cumulocity the softwares on your devices
+This document describes how to manage the software modules that are installed on a thin-edge device from the 
+cloud using the **software management** feature of thin-edge.io.
 
-# Introduction
+> Note: As of now its supported only from Cumulocity cloud. Also, only supports **debian** based
+ distributions(Ex: RaspberryPi OS , Ubuntu, Debian), which use the **apt** package manager.
 
-This document describes how to install and use the **Software management** feature in thin-edge.io.
+Three components are required on your devices to enable software management:
 
-> Note: As of now the software management only supports **debian** based distros(Ex: Ubuntu, Debian),
- which use the **apt**.
+1. Software management mapper for Cumulocity cloud
+   Sm-mapper acts  as a proxy between the cloud and the device.
+   This translates the cloud specific message type into device specific type and vice-versa.(Example: Cumulocity smart-rest to/from thin-edge json)
+   The messages from cloud will be translated and forwarded to the `tedge_agent` and messages from `tedge_agent` will be translated and sent to cumulocity cloud.
+   You can find this process with the name `tedge_mapper sm-c8y` in `ps` once it starts.
 
+2. Software management agent
+   The thin-edge software management agent is the one that calls the plugins.
+   You can find this process with the name `tedge_agent` in `ps` once it starts.
 
+3. Software management plugin
+   Plugins are the interfaces that call the package manager (example: apt/apt-get) to do the software management operations (Install, Remove or update)
+   You can find them in /etc/tedge/sm-plugins.
+   As of now there is only one (apt) plugin is supported.
```

Comment:
These 3 bullet points detailing the internal architecture: isn't this too much information for an end-user reading a tutorial? I think these process/service responsibility descriptions belong in a separate architecture/design section rather than here.

Reply:
Added as per @didier-wenzek suggestion. I feel as a dummy user, he/she should know what's happening inside the software management feature. So, felt its a good idea. Thats the difference between the howto vs tutorials.

Reply:
I put them in that order because it's easy to explain the operation flow from the cloud to the device.